### DISCUSSION TITLE
Administrative IA feedback, concerns, and open questions

### DISCUSSION THREAD

**Discussion by jordanh** [ParabolInc/parabol#5661] [Feedback]
## Issue - Enhancement

Originally #4407 

As the IT administrator of Parabol—either cloud-hosted or privately hosted—I want to:
   * configure SAML for one (or more) domains
   * bulk administer users: filter a list by search criteria bulk select to pause or remove them
      * search criteria: email, preferred name, create/last active dates, optional: team name (see #5664)
   * view an audit log of significant events: logins, logouts, billing cycle events, CRUD events for User, Org, Team, and Meeting objects
   * see overall usage statistics: number of users, active users, team names/sizes, team meeting frequency
   * enable disable integrations organization wide
   * view team details: membership, activity, integration usage

### Questions to consider during design
   * Does this design support org of orgs/team of teams? Is that a pre-requisite?
   * Are there different permission roles (billing leader vs. super admin vs. team lead vs regular user) and how do they interact with the design?
      * Note: a tension today are team leads not believing there are X teams and Y users attached to an org
   * How is this navigated to in our overall IA?

### Acceptance Criteria (optional)
   - A series of lo-fi designs
   - Feedback collected from the internal team

- **Estimated effort:** 15 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

## See also:
- #4805
- #4180
- #4321
- #4406

**Comment by gcrickman** [comment]
[A user](https://app.hubspot.com/contacts/3888472/contact/10398101) 🔒  says:

It would be nice to have different roles with different permissions ex. the Scrum Master would have administrative capabilities but there could also be a facilitator role with fewer permissions on a team or org

**Comment by gcrickman** [comment]
[A user](https://app.hubspot.com/contacts/3888472/contact/8626801) 🔒  said:

> We would like to have a dashboard with a global view on what is happening, a place where the CEO or an executive could see an overview of what's happening with all of the teams.

**Comment by gcrickman** [comment]
[A customer](https://app.hubspot.com/contacts/3888472/contact/10154751) 🔒  asks:

> Is it possible to see the member list alphabetically so it’s easier to search?

**Comment by gcrickman** [comment]
A [customer](https://app.hubspot.com/contacts/3888472/contact/3322301) 🔒  writes: 

> I have another question regarding maintenance and tracking of our usage. As a billing leader, is it possible to see the number of new teams added in our org? Or at least get some visibility on the usage of Parabol by teams? At the moment, I can only see teams that I am part of. For context, within XXXXX we've started some efforts towards promoting retrospectives on teams with the use of Parabol. Following our efforts, I'd like to be able to measure success by checking the number of teams that are added

**Comment by gcrickman** [comment]
A [customer](https://app.hubspot.com/contacts/3888472/contact/3645601) 🔒  writes in response to our existing enterprise report: 

> this report is awesome! Thank you so much for sharing! I have to admit I totally dig such graphs so I love to look deeper into such KPIs. And seeing how Parabol is being used almost on a daily basis in XXX I'm sure that it leaves no doubt to our upper management that partnering with you was a good decision. 🙂
> 
> As for stats I'd deem useful, at first glance I would probably propose:
> - The possibility to look at specific time frames (e.g. a specific month)
> - The possibility to drill deeper into the numbers, e.g. which person has how many teams and then a specific look into how the 
> meetings per team (so basically some of the info from the CSV in a more readable format)
> - Adding Estimation meetings
> 
> But even like this I'm already hooked! Not sure how much it would be used by your overall user base though, but if you offered these KPIs as a user feature for Organization Leaders I'd probably look at it on a regular basis.
> 
> Would it be cool for you if I approached you from time to time (i.e. once every six months or so) to get an updated report? Especially if I notice our user base rising over time it might come in handy to convey the need for an upgraded license in the future (btw, I finished cleaning up our user base and duplicate entries today and for now we're down to 241 actives again, so still pretty close to our max user count).

I think these suggestions would be great to think about as we design an admin board/view

**Comment by gcrickman** [comment]
A [customer](https://app.hubspot.com/contacts/3888472/contact/28771551) 🔒 writes:

> One thing to mention is that we would like to have a few people have full visibility to the history of ceremonies and it seems there's no easy way to do that but invite them to every team even if they will not participate...

**Comment by mattkrick** [comment]
datadog's SAML page is :ok_hand: 
![Screenshot from 2021-08-19 10-42-31](https://user-images.githubusercontent.com/5514175/130117984-7ae7ec65-b514-4848-9def-dfa542364a30.png)

**Comment by drewhousman1010** [comment]
A [user](https://app.hubspot.com/contacts/3888472/contact/14061851) 🔒  writes:

> I wish for better access management over the organization. Also it should be a separate entry point. Currently you organize everything over the teams, but IMO for an organization you should organize it from a central point instead. 
> 
> Also I wish for the following roles: “Member”, “Team-Admin”, “Organization-Admin”, “Billing-Leader”. 
> 
> There should be some key people of the company able to have power over everything and normal members should not be able to change anything.

**Comment by jordanh** [comment]
Our pals [at this company](https://app.hubspot.com/contacts/3888472/company/1696726269) 🔒 would like a way to map Identity Entitlements (i.e. "administrator") to a permissions level in Parabol (i.e. "Billing Leader")

**Comment by ackernaut** [comment]
A few notes here:

In Oct 2021 we had a design sprint to address big ideas and opportunities around admin. The [artifacts are here in Figma](https://www.figma.com/files/project/39111462/Administrative-IA?fuid=254682236355919698) and I’ve drafted [some notes on the process in Notion](https://www.notion.so/parabol/Administrative-IA-and-features-8443fdec91994bc2ac15d893190f6607). We focused on a goal of providing customers the information they need in order to move forward with a purchase. Today this is a manual process and one of the first leaky points today. How could we build a first set of interfaces to provide them with more of this automatically? Later we would realize what controls we need for ongoing administration of the account.

We focused on 2 questions: 1) how much domain-wide information is too much to share? and 2) what kind of basic administration needs do they expect around members (licensed seats, active or inactive) and team insights?

The list of concerns in the top comment is rather long, indeed warranting broader exploration. In [our prototype we did imagine a basic, high-level IA](https://www.figma.com/proto/u4JmGghyoLS2gHcFtmrPQG/Admin-Prototype?page-id=0%3A1&node-id=28%3A1187&viewport=501%2C48%2C0.15&scaling=scale-down&starting-point-node-id=28%3A1187) where certain concerns would be sub-views of top-level sections: Insights, Members & Teams, Plan & Billing, Advanced Settings, and Support. We were inspired by services like Google Fi that allow high-level navigation to dive down into details. We should consider expanding these sections 1 at a time with a second level nav for a handful of specific areas each, like e.g. SAML/SSO under Advanced Settings.

What I want to do here is convert this issue (originally #4407) into a 👂 Feedback discussion item, since it has a lot of verbatim from customers. And then write some smaller issues representing different areas of concern mentioned in the initial comment and takeaways from the design sprint. Many of these should be discussions or smaller, specific design scopes.

**Comment by ackernaut** [comment]
I broke this down into some issues and discussions

### Issues
- #5669 
- #5665
- #5664
- Existing related issues:
  - #4805
  - #3955

### Discussions
- #5673
- #5672
- #5671
- #5670
- #5666
- #5663
- https://github.com/ParabolInc/parabol/discussions/5675

[See notes on next steps in Notion](https://www.notion.so/parabol/Administrative-IA-and-features-8443fdec91994bc2ac15d893190f6607#90074990e85f43a5aa2e90a5345570ee)

**Reply by ackernaut** [reply]
An update here, I moved some issues into a milestone: https://github.com/ParabolInc/parabol/milestone/135

**Comment by ackernaut** [comment]
Here’s [feedback from another user](https://github.com/ParabolInc/parabol/issues/5669#issuecomment-1128124710) who wants an admin role that can see and invite to all teams on the org (implying they don’t have to be a team member to see the teams etc.)
